packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -328,8 +355,14 @@ contract Bridge is EssentialContract, IBridge {
         refundAmount += remainingFee;
 
         if (!transactedByOwner) {
-            refundAmount -= remainingFee;
-            msg.sender.sendEtherAndVerify(remainingFee, _SEND_ETHER_GAS_LIMIT);
+            uint256 fee = _calcFee(
+                _message.fee,
+                _message.gasLimit,
+                processInTheSameTx ? GAS_RECEIVING_AND_PROCESSING : GAS_PROCESSING,
```

Comment:
Seems like this does not take into account the actual gas used for processing the user's call? Don't see how a fixed gas cost could work here?

Reply:
if GAS_RECEIVING cannot be a fixed number, then I'll have to reconsider the solution.